File: AllInOne/src/TwoPointers/RemoveDupFromSortedArr.java
package src.TwoPointers;

/**
 * 
 * 26. Remove Duplicates from Sorted Array
 * 
 * @param nums
 * @return
 * 
 * @history Oct 15, 2022
 *
 */
public class RemoveDupFromSortedArr {
  public int removeDuplicates(int[] nums) {
        
    assert nums != null && nums.length >= 1 && nums.length <= 3 * Math.pow(10, 4);
    
    int preNum = nums[0];
    int validPos = 1;
    
    for (int idx = 1; idx < nums.length; idx ++) {
        
        if (preNum == nums[idx]) {
            continue;
        } else {
            nums[validPos ++] = nums[idx];
            preNum = nums[idx];
        }
    }
    
    return validPos --;
}

File: AllInOne/src/TwoPointers/RemoveDupFromSortedArr.js
/**
 * 
 * 26. Remove Duplicates from Sorted Array
 * 
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
  if (!nums) {``
    return -1;
  }   
  
  let insertIdx = 1;

  for (let idx = 1; idx < nums.length; idx ++) {
    if (nums[idx - 1] != nums[idx]) {
      nums[insertIdx] = nums[idx];
      insertIdx ++;
    }
  }

  return insertIdx;
};
